charts/k8s-monitoring/templates/alloy_config/_profiles_java.alloy.txt

```diff
@@ -0,0 +1,61 @@
+{{ define "alloy.config.profilesJava" }}
+// Profiles
+{{- if .Values.profiles.java.enabled }}
+discovery.kubernetes "java_pods" {
+  selectors {
+    role = "pod"
+    field = "spec.nodeName=" + env("HOSTNAME")
+  }
+{{- if .Values.profiles.java.namespaces }}
+  namespaces {
+    names = {{ .Values.profiles.java.namespaces | toJson }}
+  }
+{{- end }}
+  role = "pod"
+}
+
+discovery.relabel "java_pods" {
+  targets = discovery.kubernetes.java_pods.targets
+  rule {
+    action = "drop"
+    regex = "Succeeded|Failed|Completed"
+    source_labels = ["__meta_kubernetes_pod_phase"]
+  }
```

Comment:
In order to be able to get the command line executable you need to also use discovery.process with join. This is how it should look like:

```suggestion
discovery.process "java_pods" {
  join = discovery.kubernetes.java_pods.targets
}

discovery.relabel "java_pods" {
  targets = discovery.process.java_pods.targets
  rule {
    action = "drop"
    regex = "Succeeded|Failed|Completed"
    source_labels = ["__meta_kubernetes_pod_phase"]
  }
  // Filter only java processes
 rule {
        source_labels = ["__meta_process_exe"]
        action = "keep"
        regex = ".*/java$"
  }
```